snuba/admin/static/api_client.tsx

```diff
@@ -94,17 +92,19 @@ function Client() {
           })
       );
     },
-    getClickhouseCannedQueries: () => {
-      const url = baseUrl + "clickhouse_queries";
-      return fetch(url).then((resp) => resp.json());
-    },
     executeQuery: (query: QueryRequest) => {
       const url = baseUrl + "run_clickhouse_system_query";
       return fetch(url, {
         headers: { "Content-Type": "application/json" },
         method: "POST",
         body: JSON.stringify(query),
-      }).then((resp) => resp.json());
+      }).then((resp) => {
+        if (resp.ok) {
+          return resp.json();
+        } else {
+          return resp.json().then(Promise.reject.bind(Promise));
+        }
+      });
```

Comment:
This looks terrible but it seems to work. Should probably all be async/await

Reply:
I think we need to add some UI container for displaying errors in general, and then make HTTP request logic more generic so every API call writes to the same error `div` or whatever.

This is perfectly good for what we already have, though